src/babylon.scene.ts

```diff
@@ -830,29 +830,46 @@
          * @see BABYLON.Animatable
          * @see http://doc.babylonjs.com/page.php?p=22081
          */
-        public beginAnimation(target: any, from: number, to: number, loop?: boolean, speedRatio: number = 1.0, onAnimationEnd?: () => void, animatable?: Animatable): Animatable {
-
-            this.stopAnimation(target);
-
+        public beginAnimation(target: any, from: number, to: number, loop?: boolean, speedRatio: number = 1.0, onAnimationEnd?: () => void, animatable?: Animatable,transitionSpeed: number = 1.0 ): Animatable {
+    
+    
+     //////use Scene.GetCurrentAnimatableByTarget(target) to get current primary playing animatable for target;
+      
+     
+          //  this.stopAnimation(target);
+            var transitionFunction = null;
+            if(transitionSpeed < 1)  {
+                                       transitionFunction = new BABYLON.LinearFADE(transitionSpeed,this._engine);
+                                     }
+              
             if (!animatable) {
-                animatable = new Animatable(this, target, from, to, loop, speedRatio, onAnimationEnd);
+                animatable = new Animatable(this, target, from, to, loop, speedRatio, onAnimationEnd,animatable,transitionFunction);

```

Comment:
He was not as you can see on the diff:
Before=> animatable = new Animatable(this, target, from, to, loop, speedRatio, onAnimationEnd);
After=>animatable = new Animatable(this, target, from, to, loop, speedRatio, onAnimationEnd,**animatable**,transitionFunction);